.github/workflows/test-build-deploy.yml

```diff
@@ -32,6 +32,24 @@ jobs:
         run: |
           mkdir -p /go/src/github.com/grafana/mimir
           ln -s $GITHUB_WORKSPACE/* /go/src/github.com/grafana/mimir
+      - name: Get golangci-lint cache path
+        id: golangcilintcache
+        run: |
+          echo "path=$(golangci-lint cache status | grep 'Dir: ' | cut -d ' ' -f2)" >> "$GITHUB_OUTPUT"
+      - name: Cache golangci-lint cache
+        uses: actions/cache@v3
+        with:
+          key: lint-golangci-lint-${{ runner.os }}-${{ hashFiles('**/go.sum', '.golangci.yml', 'Makefile') }}
```

Comment:
Why do you hash go.sum? go.sum doesn't affect the dependency computations, only go.mod does. go.sum only serves as a collection of checksums.

Reply:
I used `go.sum` because that's what both [`actions/setup-go`](https://github.com/actions/setup-go#caching-dependency-files-and-build-outputs) and [the `actions/cache` example for Go](https://github.com/actions/cache/blob/main/examples.md#go---modules) do. 

I assume this is because `go.sum` is less likely to change if `go.mod` is modified in some way that doesn't affect the overall dependency set. For example, adding a comment to `go.mod` doesn't change the set of dependencies, nor does a dependency changing from an indirect to a direct dependency, for example.

Reply:
@charleskorn I know from before that it's a fairly widespread misunderstanding that go.sum is a dependency lockfile, though. Are you sure that the actions/setup-go and actions/cache authors have not simply fallen into this trap? That is my suspicion. I would feel safer about hashing the one file that's the source of truth wrt. the project's dependency graph: go.mod.

You can for example see [this issue](https://github.com/dependabot/dependabot-core/issues/4740) on how dependabot mistakenly bases itself on go.sum.

> For example, adding a comment to go.mod doesn't change the set of dependencies, nor does changing a dependency from an indirect to a direct dependency, for example.

Vice versa, e.g. removing a (stale) checksum from go.sum doesn't change the set of dependencies either? I still don't see why we shouldn't base the cache on the source of truth (go.mod).

Reply:
I've added `go.mod` to the list of files hashed when computing the cache key, but left `go.sum` in there as a precaution against a situation where `go.sum` is out of sync with `go.mod` and we end up caching something unexpected or incomplete.